### PR TITLE
feat: add display confirmation option to AccountCreatedEvent

### PR DESCRIPTION
**Pull request by k-g-j**
## Description

Adds the option to `displayConfirmation` to the `AccountCreatedEventStruct`

In the future, all preinstalled snaps will not go through the add account confirmation flow. However, this will be necessary not to bypass the add account confirmation screens for certain preinstalled snap scenarios (Bitcoin snap in portfolio).

### CODE REVIEW

File: src/internal/events.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { JsonStruct } from '@metamask/utils';
-import { literal, object } from 'superstruct';
```
Comment (danroc) [inline src/internal/events.ts:2]: We should use `exactOptional` here

File: src/internal/events.ts [hunk 2]
```diff
@@ -12,6 +12,10 @@ export const AccountCreatedEventStruct = object({
      * New account object.
      */
     account: KeyringAccountStruct,
+    /**
+     * Display the add account confirmation in UI.
```
Comment (danroc) [inline src/internal/events.ts:16]: Probably we should word it differently to say that it instructs, or asks, MetaMask to display or hide the confirmation, but it's not garanteed that MetaMask will honor this flag.

File: src/internal/events.ts [hunk 3]
```diff
@@ -12,6 +12,10 @@ export const AccountCreatedEventStruct = object({
      * New account object.
      */
     account: KeyringAccountStruct,
+    /**
```
Comment (danroc) [inline src/internal/events.ts:15]: (To match the other definitions)

```suggestion
    account: KeyringAccountStruct,

    /**
```

File: src/internal/events.ts (hunk 1 above)
Comment (k-g-j) [reply src/internal/events.ts:2]: Addressed here [c125539](https://github.com/MetaMask/keyring-api/pull/307/commits/c12553984bde91b656adbdc51a0b5471bc813bf2)

File: src/internal/events.ts (hunk 2 above)
Comment (k-g-j) [reply src/internal/events.ts]: Also addressed here [c125539](https://github.com/MetaMask/keyring-api/pull/307/commits/c12553984bde91b656adbdc51a0b5471bc813bf2)

File: src/internal/events.ts [hunk 4]
```diff
@@ -12,6 +13,12 @@ export const AccountCreatedEventStruct = object({
      * New account object.
      */
     account: KeyringAccountStruct,
+
+    /**
+     * Instructs MetaMask to display the add account confirmation dialog in the UI.
+     * Not guaranteed to be honored by MetaMask client.
```
Comment (ccharly) [inline src/internal/events.ts:19]: Was looking for something in JSDoc to "emphasis" this line a bit more, but couldn't find anything good...

Comment (k-g-j) [reply src/internal/events.ts]: Addressed here [3154b12](https://github.com/MetaMask/keyring-api/pull/307/commits/3154b127d5732f7882108d47d542b28a3263a77a) :)

**Review by ccharly — APPROVED** [review]
LGTM

**Pull request merged**